### PR TITLE
added ability to delete documents in level store

### LINKED ISSUE #2801: Add delete document backend logic
Sub task of https://github.com/tinacms/tinacms/issues/1974

### LINKED ISSUE #2802: Add delete document user interface
Sub task of https://github.com/tinacms/tinacms/issues/1974

### PR DESCRIPTION
**Pull request by logan-anderson**
Closes #2801 

UPDATE: I added some UI as well.

so closes #2802 and #1974 as well

### CODE REVIEW

File: packages/@tinacms/graphql/src/builder/index.ts [hunk 1]
```diff
@@ -1146,7 +1186,8 @@ export class Builder {
             return astBuilder.InputValueDefinition({
               // @ts-ignore
               name: collection.name,
-              type: await this._filterCollectionDocumentType(collection),
+              // type: await this._filterCollectionDocumentType(collection),
```
Comment (kldavis4) [inline packages/@tinacms/graphql/src/builder/index.ts:1189]: delete?

File: packages/@tinacms/graphql/src/database/index.ts [hunk 2]
```diff
@@ -507,6 +543,26 @@ export class Database {
...
+      collectionIndexDefinitions = indexDefinitions?.[collection.name]
+    }
+    await this.store.delete(filepath, {
+      collection: collection.name,
+      indexDefinitions: collectionIndexDefinitions,
+    })
+
+    if (this.store.supportsSeeding()) {
+      await this.bridge.delete(filepath)
+    }
```
Comment (kldavis4) [inline packages/@tinacms/graphql/src/database/index.ts:563]: I'm not sure if this supportsSeeding check is needed? @jeffsee55 thoughts?

File: packages/@tinacms/graphql/src/database/index.ts [hunk 3]
```diff
@@ -621,3 +677,40 @@ const _indexContent = async (
...
+const _deleteIndexContent = async (
+  database: Database,
+  documentPaths: string[],
+  collection?:
+    | CollectionFieldsWithNamespace<true>
+    | CollectionTemplatesWithNamespace<true>,
+  deleteContentItems?: boolean
+) => {
+  let seedOptions: object | undefined = undefined
+  const deleteContentItemsValue = deleteContentItems ?? false
```
Comment (kldavis4) [inline packages/@tinacms/graphql/src/database/index.ts:690]: what is this for?

File: packages/@tinacms/graphql/src/database/index.ts [hunk 4]
```diff
@@ -621,3 +677,40 @@ const _indexContent = async (
...
+
+const _deleteIndexContent = async (
+  database: Database,
+  documentPaths: string[],
+  collection?:
+    | CollectionFieldsWithNamespace<true>
+    | CollectionTemplatesWithNamespace<true>,
+  deleteContentItems?: boolean
+) => {
+  let seedOptions: object | undefined = undefined
```
Comment (kldavis4) [inline packages/@tinacms/graphql/src/database/index.ts:689]:
```suggestion
  let deleteOptions: object | undefined = undefined
```

File: packages/@tinacms/graphql/src/database/index.ts [hunk 5]
```diff
@@ -621,3 +677,40 @@ const _indexContent = async (
...
+    if (!collectionIndexDefinitions) {
+      throw new Error(`No indexDefinitions for collection ${collection.name}`)
+    }
+
+    const numIndexes = Object.keys(collectionIndexDefinitions).length
+    if (numIndexes > 20) {
+      throw new Error(
+        `A maximum of 20 indexes are allowed per field. Currently collection ${collection.name} has ${numIndexes} indexes. Add 'indexed: false' to exclude a field from indexing.`
+      )
+    }
```
Comment (kldavis4) [inline packages/@tinacms/graphql/src/database/index.ts:703]:
```suggestion
```

File: packages/@tinacms/graphql/src/database/index.ts [hunk 6]
```diff
@@ -621,3 +677,40 @@ const _indexContent = async (
...
+
+    seedOptions = {
```
Comment (kldavis4) [inline packages/@tinacms/graphql/src/database/index.ts:705]:
```suggestion
    deleteOptions = {
```

File: packages/@tinacms/graphql/src/database/index.ts [hunk 7]
```diff
@@ -621,3 +677,40 @@ const _indexContent = async (
...
+      collection: collection.name,
+      indexDefinitions: collectionIndexDefinitions,
+    }
+  }
+
+  await sequential(documentPaths, async (filepath) => {
+    if (database.store.supportsSeeding()) {
```
Comment (kldavis4) [inline packages/@tinacms/graphql/src/database/index.ts:712]: not sure again if supportsSeeding checked is needed

**Comment by kldavis4** [comment]
Tested delete in tinaadmin (and directly via postman) and it seems to be working correctly

File: packages/@tinacms/graphql/src/database/index.ts (hunk 2 above)
Comment (jeffsee55) [reply packages/@tinacms/graphql/src/database/index.ts:563]: Yeah I don't think we want this check. @logan-anderson does this all work for users who aren't using the data layer?

Comment (logan-anderson) [reply packages/@tinacms/graphql/src/database/index.ts:563]: Yes it does. I will remove it.

**Comment by logan-anderson** [comment]
@kldavis4 I updated the PR based on your comments. I also added a UI for deleting.

File: packages/@tinacms/graphql/src/database/index.ts (hunk 3 above)
Comment (logan-anderson) [reply packages/@tinacms/graphql/src/database/index.ts]: Not used. Deleted

File: .changeset/soft-cheetahs-complain.md [hunk 8]
```diff
@@ -0,0 +1,7 @@
+---
+'@tinacms/datalayer': minor
+'@tinacms/graphql': minor
```
[the remark this answers is not in the document]
Comment (kldavis4) [reply .changeset/soft-cheetahs-complain.md:3]: so it should be patch, right? https://tina.io/docs/tina-cloud/api-versioning/

**Comment by kldavis4** [comment]
> @kldavis4 I updated the PR based on your comments. I also added a UI for deleting.

cool, will try out the UI now

**Comment by kldavis4** [comment]
@logan-anderson the only thing funky about the admin ui updates is how it reloads the page after delete. Not sure how hard it would be to make it work like the add document function

File: .changeset/soft-cheetahs-complain.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (logan-anderson) [reply .changeset/soft-cheetahs-complain.md:3]: Yup!

**Review by kldavis4 — APPROVED** [review]

**Pull request merged**